src/transform/src/join_implementation.rs

```diff
@@ -193,6 +193,10 @@ mod delta_queries {
             implementation,
         } = &mut new_join
         {
+            // Delta queries are supposed to involve 2 or more inputs.
```

Comment:
I'm guessing this is here because something goes wrong without this. Can you expand the comment out to say what the function of this test and early return is? Is it that 1. without it it we would offer up a bad plan for filters, 2. without it we might crash because we now offer unary joins for implementation, or 3. we can return a bit earlier which is faster?